src/DataTable/index.jsx

```diff
@@ -107,17 +108,30 @@ function DataTable({
   // Use the state and functions returned from useTable to build your UI
   const instance = useTable(...tableArgs);
 
-  // Call ``fetchData`` whenever the state of the table changes (e.g., page change, sort or filter applied) but ignore
-  // any state changes to current row selections as we don't want to re-fetch data whenever row(s) are selected.
-  const tableStateWithoutSelections = { ...instance.state };
-  delete tableStateWithoutSelections.selectedRowIds;
+  const {
+    pageSize: tableStatePageSize,
+    pageIndex: tableStatePageIndex,
+    sortBy: tableStateSortBy,
+    filters: tableStateFilters,
+    selectedRowIds: tableStateSelectedRowIds,
+  } = instance.state;
 
   useEffect(() => {
     if (fetchData) {
-      fetchData(tableStateWithoutSelections);
+      fetchData({
+        pageSize: tableStatePageSize,
+        pageIndex: tableStatePageIndex,
+        sortBy: tableStateSortBy,
+        filters: tableStateFilters,
+      });
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [fetchData, JSON.stringify(tableStateWithoutSelections)]);
+  }, [fetchData, tableStatePageSize, tableStatePageIndex, tableStateSortBy, tableStateFilters]);
+
+  useEffect(() => {
+    if (onSelectedRowsChanged) {
+      onSelectedRowsChanged(tableStateSelectedRowIds);
```

Comment:
Inform the parent/consuming component that row selections were changed in some way.